hermes-common/src/main/java/pl/allegro/tech/hermes/infrastructure/logback/AggregatingTurboFilter.java

```diff
@@ -0,0 +1,192 @@
+            return FilterReply.NEUTRAL;
+        }
+
+        LoggingEventKey loggingEventKey = new LoggingEventKey(message, params, level, getEnrichedMarker(marker));
+        logAggregates.computeIfAbsent(logger, l -> new LoggerAggregates())
+                .aggregates.merge(loggingEventKey, new AggregateSummary(ex),
+                (currentAggregate, emptyAggregate) -> AggregateSummary.incrementCount(currentAggregate, ex));
+
+        return FilterReply.DENY;
+    }
+
+    private boolean isAggregatedLog(Marker marker) {
+        return marker != null && (marker.equals(MARKER) || marker.contains(MARKER));
+    }
+
+    private Marker getEnrichedMarker(Marker marker) {
+        if (marker == null) {
+            return MARKER;
+        }
+        marker.add(MARKER);
+        return marker;
+    }
+
+    public void addAggregatedLogger(String logger) {
+        this.aggregatedLogger.add(logger);
+    }
+
+    public long getReportingIntervalMillis() {
+        return reportingIntervalMillis;
+    }
+
+    public void setReportingIntervalMillis(long reportingIntervalMillis) {
+        this.reportingIntervalMillis = reportingIntervalMillis;
+    }
+
+    private static class LoggerAggregates {
+
+        private final Map<LoggingEventKey, AggregateSummary> aggregates = new ConcurrentHashMap<>();
+    }
+
+    private static class AggregateSummary {
+
+        private final int logsCount;
+        private final Throwable lastException;
+
+        private AggregateSummary(Throwable lastException) {
+            this(1, lastException);
+        }
+
+        private AggregateSummary(int logsCount, Throwable lastException) {
+            this.logsCount = logsCount;
+            this.lastException = lastException;
+        }
+
+        private static AggregateSummary incrementCount(AggregateSummary currentAggregate, Throwable lastException) {
+            return new AggregateSummary(currentAggregate.logsCount + 1,
+                    Optional.ofNullable(lastException).orElse(currentAggregate.lastException));
+        }
+    }
+
+    private static class LoggingEventKey {
+
+        private final String message;
+        private final Object[] params;
+        private final int level;
+        private final Marker marker;
+
+        LoggingEventKey(String message, Object[] params, Level level, Marker marker) {
+            this.message = message;
+            this.params = params;
+            this.level = Level.toLocationAwareLoggerInteger(level);
+            this.marker = marker;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            LoggingEventKey that = (LoggingEventKey) o;
+            return level == that.level &&
+                    Objects.equals(message, that.message) &&
+                    Arrays.equals(params, that.params) &&
+                    Objects.equals(marker, that.marker);
```

Comment:
Maybe this one should be before `Arrays.equals`. wdyt?

Reply:
It's IDE-generated, but yes, this would make sense.

Reply:
OK it simply depends on the order of fields, I'll fix it.

Reply:
fixed :)